File: autumn/human-readable-number/rb/while.rb
def number_format(d)
   e, f = d, 0
   while e >= 1000
      e /= 1000.0
      f += 1
   end
   return '%.3f' % e + ['', ' k', ' M', ' G'][f]
end

s = number_format(9012345678)
puts s == '9.012 G'

File: autumn/human-readable-number/rs/while.rs
fn number_format(d: f64) -> String {
   let (mut e, mut f) = (d, 0);
   while e >= 1e3 {
      e /= 1e3;
      f += 1;
   }
   format!("{:.3}", e) + ["", " k", " M", " G"][f]
}

fn main() {
   let s = number_format(9012345678.0);
   println!("{}", s == "9.012 G");
}
